### ISSUE TITLE
RuleSet.toCNF hanging with expression

### ISSUE THREAD

**Issue by mbomio** [bpodgursky/jbool_expressions#8]
Hi,

When calling RuleSet.toCNF with the  following expression: 
```
((105177 | 12030) & ((105098 | 105097 | 105096 | 105095 | 12003 | 12002 | 12001 | 12000) & ((12096 | 106083 | 106045) & 82095)))
```
the process hangs. 
Here's my thread dump:
```
   java.lang.Thread.State: RUNNABLE
	at com.bpodgursky.jbool_expressions.NExpression.<init>(NExpression.java:19)
	at com.bpodgursky.jbool_expressions.And.<init>(And.java:15)
	at com.bpodgursky.jbool_expressions.And.createInternal(And.java:20)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:29)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:42)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.NExpression.apply(NExpression.java:27)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAllSingle(RuleSet.java:53)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applyAll(RuleSet.java:46)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.applySet(RuleSet.java:102)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.toSop(RuleSet.java:79)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.toPos(RuleSet.java:86)
	at com.bpodgursky.jbool_expressions.rules.RuleSet.toCNF(RuleSet.java:75)
```

Thanks

**Comment by bpodgursky** [comment]
That's strange.  I'm going to try to look into it this weekend.

**Comment by mbomio** [comment]
Thanks!

**Comment by bpodgursky** [comment]
I * think * this change will fix the problem: https://github.com/bpodgursky/jbool_expressions/commit/50f090d4347c9a38f9093bc3154a7db860f70876.  

Could you try it out and make sure it works for you?  Want to make sure I don't introduce a bug where expressions aren't always reduced down.  Do you need me to push out a snapshot, or are you building the project yourselves?

**Comment by mbomio** [comment]
Thanks for the fix! Tried the new version and it worked. Is it possible to do release with the fix?

**Comment by bpodgursky** [comment]
Great.  I've released version 1.8; should show up in central in a few hours.  Going to close this out, let me know if you see any problems.
